### ISSUE TITLE
Store|Woo: Add WooCommerce Menu Item for Business Plan/Store Users

### ISSUE THREAD

**Issue by timmyc** [Automattic/wp-calypso#48027]
**Important** Place this behind the "store-pre-sunset" feature flag (implementation tbd)

> This issue is part of a project to migrate users of the Store on WordPress.com feature ( The Woo UI in calypso ) to use the default WooCommerce experience in wp-admin instead. Further details & discussion: pbIJXs-sE-p2

Starting during the four week transition period for Store on WordPress.com users, a new menu item in the Calypso sidebar that allows users to access the WooCommerce interface on wp-admin will be added. This will allow users to easily access both the standard WooCommerce UI in wp-admin ( which they will be using after the store ui is turned off ), and the existing UI in Calypso:

<img width="1078" alt="calypso-menu-store" src="https://user-images.githubusercontent.com/22080/101104501-22270280-3580-11eb-8ebf-d2976408b2fb.png">

The current logic to create the "Store" sidebar menu item [can be found here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/sidebar/index.jsx#L661-L690)

Seems like we could create a new method like `store()` there, but instead call it `woo()` but leverage much of the existing logic checks in the existing store method. The menu item should be titled `WooCommerce`, and let's use the same shopping cart gridicon for now ( cc @elizaan36 ) as I'm not sure if we have access to a small Woo logo in the context of Calypso... though that would be really nice to have.

The link target should be the WooCommerce home screen, and it should also have the external link icon since it is just like the link to wp-admin lower in the calypso sidebar menu.

The new "WooCommerce" menu item will continue to exist once the transition is complete (the existing "Store" menu item will be removed).

~@pmcpinto are you interested at all in collecting a track when this link is clicked?~

Also collect tracks data when the WooCommerce menu item is clicked.

**Comment by pmcpinto** [comment]
> @pmcpinto are you interested at all in collecting a track when this link is clicked?

Yep, that would be great.

So during the transition, we'd have the Store menu item and the Woo menu item, correct? I'm wondering if it makes sense to add something else to the Store menu item to inform that it will be deprecated. Eg: Store (will be removed on Feb xx). We need to consider the fact that not all the Store users are going to visit the store interface during those 4 weeks. Thoughts?

**Comment by timmyc** [comment]
> So during the transition, we'd have the Store menu item and the Woo menu item, correct?

Correct. I'm thinking by having the new menu item show up as WooCommerce will align quite nicely with the further menu/nav changes coming on WPCOM... with that the eventual way the menu item will appear in calypso for these users will be "WooCommerce" since it is based off the menu registration that happens in wp-admin.

**Comment by elizaan36** [comment]
Thanks for putting these issues together @timmyc

> I'm wondering if it makes sense to add something else to the Store menu item to inform that it will be deprecated. Eg: Store (will be removed on Feb xx)

@pmcpinto maybe a tooltip would work on the Store item in the nav.
<img width="507" alt="Screen Shot 2020-12-07 at 2 14 57 PM" src="https://user-images.githubusercontent.com/5121465/101436277-7a893780-38d3-11eb-9048-3c64bda461a7.png">

**Comment by becdetat** [comment]
@timmyc @pmcpinto 

> > @pmcpinto are you interested at all in collecting a track when this link is clicked?

> Yep, that would be great.

I'm going to split this off to a separate ticket (https://github.com/Automattic/wp-calypso/issues/48155), as Calypso doesn't have tracks support and it will need to be copied in from WC.

**Comment by mattsherman** [comment]
> I'm going to split this off to a separate ticket (#48155), as Calypso doesn't have tracks support and it will need to be copied in from WC.

Calypso has Tracks support:

https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-analytics/README.md

I would think that will suit our needs. But, there is also a "Store on WordPress.com" flavor that could be used...

https://github.com/Automattic/wp-calypso/tree/trunk/client/extensions/woocommerce/lib/analytics 

I guess it depends on how we want our event named and what other properties we want on it. But, in any case, I would think one of these would be what we should use.

**Comment by becdetat** [comment]
Nice! I'll un-split this. Thanks for the catch @mattsherman

**Comment by mattsherman** [comment]
> During the four week transition period for Store on WordPress.com users, we would like to have a new menu item in the Calypso sidebar that allows users to access the WooCommerce interface on wp-admin.

What happens after the transition period? How does a user then access WooCommerce from Calypso? During the transition, they were able to easily get there by clicking this menu item. Now what? We should either keep this menu item, or teach users the method that they will eventually be using to get to WooCommerce.

**Comment by becdetat** [comment]
@mattsherman 

> How does a user then access WooCommerce from Calypso?

The WooCommerce menu item is kept post-sunset, the Store menu item will be removed on sunset.

**Comment by becdetat** [comment]
@pmcpinto what tracks data do you want sent when the button is clicked? And what should the name be?

**Comment by pmcpinto** [comment]
> @pmcpinto what tracks data do you want sent when the button is clicked? And what should the name be?

I think it makes sense to use an event naming similar to the existing events in the Store UI. This is the one that's tracking the clicks in the existing menu item: `calypso_woocommerce_store_nav_item_click`

I'd go with something like: `calypso_woocommerce_store_woo_core_item_click`

**Comment by mattsherman** [comment]
> The WooCommerce menu item is kept post-sunset, the Store menu item will be removed on sunset.

Thanks. I've updated the description to make that clear.

**Comment by ilyasfoo** [comment]
@timmyc or @pmcpinto can you help to confirm the following:

- This project will not impact existing E-commerce users as they're already accessing WooCommerce interface. On that note, it might also be important that E-commerce users **should not** see the Woo menu - because it's not relevant to them.
- To standardize users' experience, do we need to calypsoify all our links as well?

**Comment by timmyc** [comment]
Really good question @ilyasfoo - as [there is some logic in the calypso menu specifically for eCommerce plan users](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/sidebar/index.jsx#L677-L679).

So the existing logic there will render the Store menu item for ecomm plan users, pointed at their wp-admin instance. That logic needs to remain during, and post sunset of Store on WordPress.com.

> To standardize users' experience, do we need to calypsoify all our links as well?

No we are actually going to be removing calypsoify usage soon, so no need for the param.

> it might also be important that E-commerce users should not see the Woo menu - because it's not relevant to them

I think that is the correct path for the time being. As the nav unification work continues to happen on WordPress.com though, we might want to consider having any business plan, or ecommerce plan site all use a singular menu item called WooCommerce. Ultimately the nav unification work will be rendering the WooCommerce menu item in Calypso because it will show any registered menu items from wp-admin eventually. So I think we will get to the point where eCom users will have a WooCommerce menu item... but think its a bit out of scope for this issue.

**Comment by mattsherman** [comment]
> The link target should be the WooCommerce home screen, and it should also have the external link icon since it is just like the link to wp-admin lower in the calypso sidebar menu.

@timmyc Should this link also open in a new tab, as the wp-admin link does? I'd say that the two need to go together... either it shows the external link icon and opens in a new tab, or it does not show the external link icon and opens in the same tab (this is how things currently work with the external link icon in Calypso).

**Comment by mattsherman** [comment]
> The menu item should be titled WooCommerce, and let's use the same shopping cart gridicon for now ( cc @elizaan36 ) as I'm not sure if we have access to a small Woo logo in the context of Calypso... though that would be really nice to have.

@elizaan36 Would you happen to have an SVG Woo logo that we could use similar to how the Jetpack logo is used in Calypso? (https://github.com/Automattic/wp-calypso/tree/trunk/client/components/jetpack-logo)

**Comment by mattsherman** [comment]
> I think that is the correct path for the time being. As the nav unification work continues to happen on WordPress.com though, we might want to consider having any business plan, or ecommerce plan site all use a singular menu item called WooCommerce. Ultimately the nav unification work will be rendering the WooCommerce menu item in Calypso because it will show any registered menu items from wp-admin eventually. So I think we will get to the point where eCom users will have a WooCommerce menu item... but think its a bit out of scope for this issue.

It's unfortunate that for now we will have a different experience for the same functionality in Business and eCommerce. Could we consider renaming "Store" to "WooCommerce" for eCommerce plan sites, with some sort of a tooltip or notice to let users know that this has been renamed? If we did this, it would be in a followup issue. /cc @elizaan36

**Comment by elizaan36** [comment]
> Would you happen to have an SVG Woo logo that we could use similar to how the Jetpack logo is used in Calypso?

Hey @mattsherman , let me know if this Woo logo SVG works. 
[WooLogo.svg.zip](https://github.com/Automattic/wp-calypso/files/5691278/WooLogo.svg.zip)

> Could we consider renaming "Store" to "WooCommerce" for eCommerce plan sites, with some sort of a tooltip or notice to let users know that this has been renamed?

We're not deprecating the "Store" area for the Ecom plan though. I think changing link to "WooCommerce" in the Ecom plan wouldn't provide much benefit to the user as it feels more integrated to call it Store.

As one solution to keep them consistent, we could keep the menu item `Store` with the Woo logo. This may be enough info to help people to understand what's happening, and would only appear during the sunset period.

<img width="602" alt="Screen Shot 2020-12-14 at 2 05 31 PM" src="https://user-images.githubusercontent.com/5121465/102129741-79956000-3e15-11eb-9f57-36a89e5c23ae.png">

What do you think?

**Comment by mattsherman** [comment]
> Thanks for putting these issues together @timmyc
> 
> > I'm wondering if it makes sense to add something else to the Store menu item to inform that it will be deprecated. Eg: Store (will be removed on Feb xx)
> 
> @pmcpinto maybe a tooltip would work on the Store item in the nav.
> <img alt="Screen Shot 2020-12-07 at 2 14 57 PM" width="507" src="https://user-images.githubusercontent.com/5121465/101436277-7a893780-38d3-11eb-9048-3c64bda461a7.png">

I've split this into a separate issue (https://github.com/Automattic/wp-calypso/issues/48373)

**Comment by mattsherman** [comment]
> Hey @mattsherman , let me know if this Woo logo SVG works.
> [WooLogo.svg.zip](https://github.com/Automattic/wp-calypso/files/5691278/WooLogo.svg.zip)

I think that SVG will work -- thanks!

> 
> > Could we consider renaming "Store" to "WooCommerce" for eCommerce plan sites, with some sort of a tooltip or notice to let users know that this has been renamed?
> 
> We're not deprecating the "Store" area for the Ecom plan though. I think changing link to "WooCommerce" in the Ecom plan wouldn't provide much benefit to the user as it feels more integrated to call it Store.
> 
> As one solution to keep them consistent, we could keep the menu item `Store` with the Woo logo. This may be enough info to help people to understand what's happening, and would only appear during the sunset period.
> 
> <img alt="Screen Shot 2020-12-14 at 2 05 31 PM" width="602" src="https://user-images.githubusercontent.com/5121465/102129741-79956000-3e15-11eb-9f57-36a89e5c23ae.png">
> 
> What do you think?

I think that ideally, when the experience is the same in the Business and eCommerce plans, it should be labelled and presented the same. Having the same thing labelled differently is confusing, especially in the case where a site starts off on the Business plan and then upgrades to the eCommerce plan (not sure how often that happens, though).

Whether it is labelled "WooCommerce" or "Store"? I see pros and cons to both. I personally like "WooCommerce" because it brings our branding in there. But, keeping "Store" has the benefit of continuity for eCommerce plan users, since as you say, their experience isn't changing.

The solution of the Woo logo with "Store" could be a good medium, though I'm not sure about two menu items labelled "Store" in the deprecation period.

I'll hold off on creating a followup issue to change the logo until the naming is finalized as well, as both things should be tackled in a single issue.

**Comment by elizaan36** [comment]
This feels like a question that's beyond UX. More around the messaging/branding of Woo within WPCOM. cc @pmcpinto or @claraclee if you have thoughts here.

**Comment by pmcpinto** [comment]
> This feels like a question that's beyond UX. More around the messaging/branding of Woo within WPCOM. cc @pmcpinto or @claraclee if you have thoughts here.

I'd go with WooCommerce:
- We mention "powered by Woo" in several places, our inbox notifications talk about WooCommerce...
- About 70-80% of eCommerce plan customers use Woo through wp-admin, so they're already used to the Woo branding in the menu.
- We want them to purchase extensions in the Woo marketplace instead of going to other places. The branding connection between the product and the marketplace can increase confidence in purchasing extensions there.
